plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java

```diff
@@ -218,9 +214,10 @@ public void onCommitClicked() {
   }
 
   private Path[] getFilesToCommitArray() {
-    Path[] filesToCommitArray = new Path[filesToCommit.size()];
-    filesToCommit.forEach(
-        file -> filesToCommitArray[filesToCommit.indexOf(file)] = Path.valueOf(file));
+    List<String> selectedFiles = selectableChangesPanelPresenter.getSelectedFiles();
+    Path[] filesToCommitArray = new Path[selectedFiles.size()];
+    selectedFiles.forEach(
```

Comment:
Use `for` loop here to avoid `selectedFiles.indexOf(file)`

Reply:
done